### PR TITLE
replaceExisting to append query parameters and delete all query paramters

### PR DESCRIPTION
**Pull request by tospery**
1. Add replaceExisting to appendingQueryParameters/appendQueryParameters.
2. Add deletingAllQueryParameters/deleteAllQueryParameters methods support.

### CODE REVIEW

File: Tests/FoundationTests/URLExtensionsTests.swift [hunk 1]
```diff
@@ -20,11 +20,16 @@ final class URLExtensionsTests: XCTestCase {
...
+		let replaceURL = URL(string: "https://www.google.com?q=swift%20swifter")!
+		XCTAssertEqual(replaceURL.appendingQueryParameters(params, replaceExisting: true), queryUrl)
     }
 
     func testAppendQueryParameters() {
         url.appendQueryParameters(params)
         XCTAssertEqual(url, queryUrl)
+		let replaceURL = URL(string: "https://www.google.com?q=swift%20swifter")!
+		replaceURL.appendQueryParameters(params, replaceExisting: true)
+		XCTAssertEqual(replaceURL, queryUrl)
```
Comment (LucianoPAlmeida) [inline Tests/FoundationTests/URLExtensionsTests.swift:32]: Can we add a test where there is an additional param added as well, only this `q` param makes function seem a no-op.

File: Sources/SwifterSwift/Foundation/URLExtensions.swift [hunk 2]
```diff
@@ -101,6 +114,29 @@ public extension URL {
...
+        var urlComponents = URLComponents(url: self, resolvingAgainstBaseURL: true)!
+        urlComponents.queryItems = nil
+        return urlComponents.url!
+    }
+    
+	/// SwifterSwift: Remove all the query parameters from the URL.
+	///
+	///        var url = URL(string: "https://domain.com?query=true")!
+	///        url.deleteAllQueryParameters()
+	///        print(url) // prints "https://domain.com"
```
Comment (LucianoPAlmeida) [inline Sources/SwifterSwift/Foundation/URLExtensions.swift:133]:
```suggestion
	///
	///    var url = URL(string: "https://domain.com?query=true")!
	///    url.deleteAllQueryParameters()
	///    print(url) // prints "https://domain.com"
```

File: Tests/FoundationTests/URLExtensionsTests.swift (hunk 1 above)
Comment (tospery) [reply Tests/FoundationTests/URLExtensionsTests.swift]: Done

File: Sources/SwifterSwift/Foundation/URLExtensions.swift (hunk 2 above)
Comment (tospery) [reply Sources/SwifterSwift/Foundation/URLExtensions.swift]: Done

**Review by LucianoPAlmeida — APPROVED** [review]

File: Sources/SwifterSwift/Foundation/URLExtensions.swift [hunk 3]
```diff
@@ -42,12 +42,24 @@ public extension URL {
     ///		url.appendingQueryParameters(params) -> "https://google.com?q=Swifter%20Swift"
     ///
     /// - Parameter parameters: parameters dictionary.
+	/// - Parameter replaceExisting: whether replace existing query parameters or not.
     /// - Returns: URL with appending given query parameters.
-    func appendingQueryParameters(_ parameters: [String: String]) -> URL {
+    func appendingQueryParameters(_ parameters: [String: String], replaceExisting: Bool = false) -> URL {
```
Comment (guykogus) [inline Sources/SwifterSwift/Foundation/URLExtensions.swift:47]: This should be done in a separate PR. Each PR should be for a specific change.

File: Sources/SwifterSwift/Foundation/URLExtensions.swift [hunk 4]
```diff
@@ -59,8 +71,9 @@ public extension URL {
     ///		print(url) // prints "https://google.com?q=Swifter%20Swift"
     ///
     /// - Parameter parameters: parameters dictionary.
-    mutating func appendQueryParameters(_ parameters: [String: String]) {
-        self = appendingQueryParameters(parameters)
+	/// - Parameter replaceExisting: whether replace existing query parameters or not.
+    mutating func appendQueryParameters(_ parameters: [String: String], replaceExisting: Bool = false) {
```
Comment (guykogus) [inline Sources/SwifterSwift/Foundation/URLExtensions.swift:75]: Same, separate PR

File: CHANGELOG.md [hunk 5]
```diff
@@ -45,6 +47,8 @@ The changelog for **SwifterSwift**. Also see the [releases](https://github.com/S
   - Refactored `tint(_:blendMode:)` using UIGraphicsImageRenderer if available. [#731](https://github.com/SwifterSwift/SwifterSwift/pull/731) by [FraDeliro](https://github.com/FraDeliro).
 - **Sequence**:
   - Corrected documentation for `sorted(by:with:)` and `sorted(by:)`. [#751](https://github.com/SwifterSwift/SwifterSwift/pull/751) by [mmdock](https://github.com/mmdock)
+- **URL**:
+  - Updated `appendingQueryParameters(_:replaceExisting:)` and `appendQueryParameters(_:replaceExisting:)`. [#1193](https://github.com/SwifterSwift/SwifterSwift/pull/1193) by [tospery](https://github.com/tospery)
```
Comment (guykogus) [inline CHANGELOG.md:51]: Separate PR

**Review by guykogus — CHANGES_REQUESTED** [review]

**Comment by guykogus** [comment]
Closing as stale. Please reopen if you wish to continue.
